py-polars/polars/expr/expr.py

```diff
@@ -3517,7 +3517,8 @@ def tail(self, n: int = 10) -> Self:
         └─────┘
 
         """
-        return self._from_pyexpr(self._pyexpr.tail(n))
+        offset = -expr_to_lit_or_expr(n, str_to_lit=False)
+        return self.slice(offset, n)
 
     def limit(self, n: int = 10) -> Self:
```

Comment:
`limit` then also needs to be updated - it's an alias for `head`:

```suggestion
    def limit(self, n: int | Expr = 10) -> Self:
```